type.py

```diff
@@ -39,6 +40,7 @@ class BaseConfig:
     force_fit: bool
     save: bool
     save_remote: bool
+    preferred_load_origin: Literal[Const.remote, Const.local, Const.pretrained, None]
```

Comment:
it's a lot easier if it's an enum, you don't need to manually list it everywhere!